src/m365/yammer/commands/yammer-search.ts

```diff
@@ -0,0 +1,214 @@
+              continueprocessing = false;
+              this.messages = this.messages.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultTopics.length > 0) {
+            resultsFound = true;
+            this.topics = this.topics.concat(resultTopics);
+
+            if (args.options.limit && this.topics.length > args.options.limit) {
+              continueprocessing = false;
+              this.topics = this.topics.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultGroups.length > 0) {
+            resultsFound = true;
+            this.groups = this.groups.concat(resultGroups);
+
+            if (args.options.limit && this.groups.length > args.options.limit) {
+              continueprocessing = false;
+              this.groups = this.groups.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultUsers.length > 0) {
+            resultsFound = true;
+            this.users = this.users.concat(resultUsers);
+
+            if (args.options.limit && this.users.length > args.options.limit) {
+              continueprocessing = false;
+              this.users = this.users.slice(0, args.options.limit);
+            }
+          }
+
+          // does not process more queries if we are not able to return a complex object 
+          if (resultsFound && continueprocessing && args.options.output === 'json') {
+            this
+                .getAllItems(logger, args, ++page)
+                .then((): void => {
+                  resolve();
+                }, (err: any): void => {
+                  reject(err);
+                });
+          }
+          else {
+            resolve();
+          }
+        }, (err: any): void => {
+          reject(err);
+        });
+    });
+  };
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this.summary = {
+      messages: 0,
+      groups: 0,
+      topics: 0,
+      users: 0
+    }
+    this.messages = [];
+    this.groups = [];
+    this.topics = [];
+    this.users = [];
+
+    this
+      .getAllItems(logger, args, 1)
+      .then((): void => {
+        if (args.options.output === 'json') {
+          logger.log(
+            {
+              summary: this.summary,
+              messages: this.messages,
+              users: this.users,
+              topics: this.topics,
+              groups: this.groups,
+            });
+        }
+        else {
+          logger.log(this.summary);
+        }
+
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  };
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-s, --search <search>',
+        description: 'The query for the search'
+      },
+      {
+        option: '--limit [limit]',
+        description: 'Limits the results returned for each item category. Can only be used with the --output json option.'
```

Comment:
If we adjust the output to present the results, then the limit option should apply in all output modes.

Reply:
See comment above. After we decided we can close this one

Reply:
Added the show command option to improve the handling